Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool.hs

```diff
@@ -425,6 +428,153 @@ calculateDriverPoolWithActualDist poolCalculationStage driverPoolCfg mbVariant p
+        { distanceToPickup = distanceToDriver,
+          ..
+        }
+
+calculateDriverCurrentlyOnRideWithActualDist ::
+  ( EncFlow m r,
+    CacheFlow m r,
+    EsqDBFlow m r,
+    Esq.EsqDBReplicaFlow m r,
+    CoreMetrics m,
+    HasDriverPoolConfig r,
+    HasCoordinates a
+  ) =>
+  PoolCalculationStage ->
+  DriverPoolConfig ->
+  Maybe Variant ->
+  a ->
+  Id DM.Merchant ->
+  Maybe PoolRadiusStep ->
+  Int ->
+  m [DriverPoolWithActualDistResult]
+calculateDriverCurrentlyOnRideWithActualDist poolCalculationStage driverPoolCfg mbVariant pickup merchantId mRadiusStep reduceRadiousValue = do
+  driverPool <- calculateDriverPoolCurrentlyOnRide poolCalculationStage driverPoolCfg mbVariant pickup merchantId mRadiusStep reduceRadiousValue
+  case driverPool of
+    [] -> return []
+    (a : pprox) -> do
+      let y = driverResultFromDestinationLocation <$> (a :| pprox)
+      driverPoolWithActualDistFromDestinationLocation <- computeActualDistance merchantId pickup y
+      driverToDestinationDistanceThreshold <- asks (.driverPoolCfg.driverToDestinationDistanceThreshold)
+      driverPoolWithACtualDistFromCurrentLocation <- sequence $ calculateActualDistanceCurrently driverToDestinationDistanceThreshold <$> (a :| pprox)
+      let driverPoolWithActualDist = NE.fromList (zipWith (curry combine) (NE.toList driverPoolWithActualDistFromDestinationLocation) (NE.toList driverPoolWithACtualDistFromCurrentLocation))
+      let filtDriverPoolWithActualDist = case driverPoolCfg.actualDistanceThreshold of
+            Nothing -> NE.toList driverPoolWithActualDist
+            Just threshold -> NE.filter (filterFunc threshold) driverPoolWithActualDist
+      logDebug $ "secondly filtered driver pool result currently on ride" <> show filtDriverPoolWithActualDist
+      return filtDriverPoolWithActualDist
+  where
+    filterFunc threshold estDist = getMeters estDist.actualDistanceToPickup <= fromIntegral threshold
+    driverResultFromDestinationLocation DriverPoolResultCurrentlyOnRide {..} =
+      DriverPoolResult
+        { driverId = driverId,
+          language = language,
+          driverDeviceToken = driverDeviceToken,
+          distanceToPickup = distanceToPickup,
+          variant = variant,
+          lat = destinationLat,
+          lon = destinationLon
+        }
+    calculateActualDistanceCurrently driverToDestinationDistanceThreshold DriverPoolResultCurrentlyOnRide {..} = do
+      let temp =
+            DriverPoolResult
+              { driverId = driverId,
+                language = language,
+                driverDeviceToken = driverDeviceToken,
+                distanceToPickup = distanceToPickup,
+                variant = variant,
+                lat = lat,
+                lon = lon
+              }
+      if distanceFromDriverToDestination < driverToDestinationDistanceThreshold
+        then do
+          defaultPopupDelay <- asks (.defaultPopupDelay)
+          time <- asks (.driverPoolCfg.driverToDestinationDuration)
```

Comment:
use `driverPoolConfig` that is passed as argument in the function instead of `asks (.driverPoolConfig)`